### PR TITLE
i18n(ja): Add sidebar.mdx

### PR DESCRIPTION
**Pull request by morinokami**
#### What kind of changes does this PR include?

- Changes or translations of Starlight docs site content

#### Description

- Add sidebar.mdx.
  - The diffs in https://github.com/withastro/starlight/pull/696 are already included (so this PR should be merged after that PR has been merged).

### CODE REVIEW

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+
+### リンク
+
+`label`と`link`プロパティをもつオブジェクトを使用して、内部または外部ページへのリンクを追加します。
+
+```js
+starlight({
+	sidebar: [
+		// CSSとスタイリングガイドへのリンク。
+		{ label: 'CSSとスタイリング', link: '/ja/guides/css-and-tailwind/' },
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:68]: I'm not 100% sure how links in examples should be handled. However, I looked at the italian translation and found that all examples link to englisch content. Therefore, I _think_ you need to make this little improvement on every sidebar example:

```suggestion
		{ label: 'CSSとスタイリング', link: '/guides/css-and-tailwind/' },
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+		{ label: 'Astro', link: 'https://astro.build/' },
+	],
+});
+```
+
+上の設定により、以下のサイドバーが生成されます。
+
+<SidebarPreview
+	config={[
+		{ label: 'CSSとスタイリング', link: '/ja/guides/css-and-tailwind/' },
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:79]:
```suggestion
		{ label: 'CSSとスタイリング', link: '/guides/css-and-tailwind/' },
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+`label`と`items`プロパティをもつオブジェクトを使用して、グループを追加します。`label`はグループの見出しとして使用されます。`items`配列にリンクまたはサブグループを追加します。
+
+```js
+starlight({
+	sidebar: [
+		// 「ガイド」というラベルのリンクグループ。
+		{
+			label: 'ガイド',
+			items: [
+				{ label: 'コンポーネント', link: '/ja/guides/components/' },
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:97]:
```suggestion
				{ label: 'コンポーネント', link: '/guides/components/' },
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+				{ label: '国際化（i18n）', link: '/ja/guides/i18n/' },
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:98]:
```suggestion
				{ label: '国際化（i18n）', link: '/guides/i18n/' },
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+```
+
+上の設定により、以下のサイドバーが生成されます。
+
+<SidebarPreview
+	config={[
+		{
+			label: 'ガイド',
+			items: [
+				{ label: 'コンポーネント', link: '/ja/guides/components/' },
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:121]:
```suggestion
				{ label: 'コンポーネント', link: '/guides/components/' },
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+				{ label: '国際化（i18n）', link: '/ja/guides/i18n/' },
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:122]:
```suggestion
				{ label: '国際化（i18n）', link: '/guides/i18n/' },
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 7]
```diff
@@ -0,0 +1,467 @@
...
+				{
+					label: 'スタイリング',
+					items: [
+						{ label: 'CSS', link: '/ja/guides/css-and-tailwind/' },
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:126]:
```suggestion
						{ label: 'CSS', link: '/guides/css-and-tailwind/' },
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 8]
```diff
@@ -0,0 +1,467 @@
...
+						{ label: 'Tailwind', link: '/ja/guides/css-and-tailwind/' },
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:127]:
```suggestion
						{ label: 'Tailwind', link: '/guides/css-and-tailwind/' },
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 9]
```diff
@@ -0,0 +1,467 @@
...
+						{ label: 'Shiki', link: '/ja/guides/css-and-tailwind/' },
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:128]:
```suggestion
						{ label: 'Shiki', link: '/guides/css-and-tailwind/' },
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 10]
```diff
@@ -0,0 +1,467 @@
...
+```js
+starlight({
+	sidebar: [
+		{
+			label: 'ガイド',
+			items: [
+				// 「新規」バッジ付きのリンク。
+				{
+					label: 'コンポーネント',
+					link: '/ja/guides/components/',
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:246]:
```suggestion
					link: '/guides/components/',
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 11]
```diff
@@ -0,0 +1,467 @@
...
+上の設定により、以下のサイドバーが生成されます。
+
+<SidebarPreview
+	config={[
+		{
+			label: 'ガイド',
+			items: [
+				{
+					label: 'コンポーネント',
+					link: '/ja/guides/components/',
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:264]:
```suggestion
					link: '/guides/components/',
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 12]
```diff
@@ -0,0 +1,467 @@
...
+```js
+starlight({
+	sidebar: [
+		{
+			label: 'ガイド',
+			items: [
+				// 「実験的」バッジ付きのリンク。
+				{
+					label: 'コンポーネント',
+					link: '/ja/guides/components/',
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:287]:
```suggestion
					link: '/guides/components/',
```

File: docs/src/content/docs/ja/guides/sidebar.mdx [hunk 13]
```diff
@@ -0,0 +1,467 @@
...
+上の設定により、以下のサイドバーが生成されます。
+
+<SidebarPreview
+	config={[
+		{
+			label: 'ガイド',
+			items: [
+				{
+					label: 'コンポーネント',
+					link: '/ja/guides/components/',
```
Comment (trueberryless) [inline docs/src/content/docs/ja/guides/sidebar.mdx:305]:
```suggestion
					link: '/guides/components/',
```

**Review by trueberryless — CHANGES_REQUESTED** [review]
Thank you, @morinokami! I'm not 100% sure how links in examples should be handled. However, I looked at the italian translation and found that all examples link to englisch content. Therefore, I _think_ you need to make these little improvements on every sidebar example.

As I said, I'm not sure if I'm correct. Please inform me if you know more than me...

**Review by delucis — APPROVED** [review]
Thank you for this translation @morinokami and for the review @trueberryless!

I accepted the suggestions inside the code examples but not inside the `<SidebarPreview>` components. My reasoning here is that in your Starlight config you usually do not always include the language segment in links (e.g. `/ja/`). But Starlight will add the language segment for you if needed, so our preview components can still show the `/ja/` links.

**Pull request merged**